test/install_test.go

```diff
@@ -247,20 +247,29 @@ func TestInstallOrUpgradeCli(t *testing.T) {
 
 	// test `linkerd upgrade --from-manifests`
 	if TestHelper.UpgradeFromVersion() != "" {
-		manifests, err := TestHelper.Kubectl("",
-			"--namespace", TestHelper.GetLinkerdNamespace(),
-			"get", "configmaps/"+k8s.ConfigConfigMapName, "secrets/"+k8s.IdentityIssuerSecretName,
-			"-oyaml",
-		)
+		resources := []string{}
+		resources = append(resources, "configmaps/"+k8s.ConfigConfigMapName, "secrets/"+k8s.IdentityIssuerSecretName)
+
+		// If `linkerd-config-addons` exists, add it to the resources to get
+		_, err := TestHelper.Kubectl("", "--namespace", TestHelper.GetLinkerdNamespace(), "get", "configmaps/"+k8s.AddOnsConfigMapName)
+		if err != nil {
+			resources = append(resources, "configmaps/"+k8s.AddOnsConfigMapName)
+		}
+
+		args := append([]string{"--namespace", TestHelper.GetLinkerdNamespace(), "get"}, resources...)
+		args = append(args, "-o yaml")
```

Comment:
Maybe the test is failing also because `-o yaml` is used in a single arg?

Reply:
Yes definitely it. Thanks!